doc/status-code/api/status-codes.rst

```diff
@@ -205,7 +205,9 @@ Error codes
 
    .. summary:: A status code that indicates that a handle parameter is not valid.
 
-   A function can return this error any time a handle parameter is invalid.
+   A function can return this error any time it expects a parameter to be a handle to an existing object, but the handle is invalid.
+
+   This status code is not intended for cases where the parameter is a value that is chosen by the caller, but the value does not satisfy a validity condition for handles of this kind. In such cases, it is recommended that a function returns `PSA_ERROR_ALREADY_EXISTS` if value is valid but already in use, and `PSA_ERROR_INVALID_ARGUMENT` or a more specific status code if the value is otherwise invalid.
```

Comment:
With the change to the existing paragraph, I am not convinced that this is needed? - the new scoping in para 1 clearly indicates that this return code is only applicable for APIs where the 'handle' parameter is meant to refer to an existing item.

APIs in which the caller is requesting a specific value for a new handle are not in that scope.

If we do want to keep this, I find the wording of the first sentence unclear. Would this be any better:

> This status code is not recommended when a caller is requesting a specific handle value to be used for a new handle.